jctools-core/src/test/java/org/jctools/queues/atomic/ScLinkedAtomicQueueTest.java

```diff
@@ -0,0 +1,75 @@
+package org.jctools.queues.atomic;
+
+import org.junit.Test;
+
+import java.util.Queue;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+public abstract class ScLinkedAtomicQueueTest {
+    protected abstract Queue<Integer> newQueue();
+
+    @Test
+    public void removeWorksInBeginMiddleEnd() throws InterruptedException {
```

Comment:
@nitsanw - I didn't integrate this with the typical `QueueSanityTest` and `ConcurrentQueueSpec` approach because not all queues support this feature yet and figured it would be better to keep independent for now. Let me know if you have a more preferred approach.